eventual-cassandra/src/main/scala/com/evolutiongaming/kafka/journal/eventual/cassandra/ReplicatedCassandra.scala

```diff
@@ -443,10 +447,15 @@ private[journal] object ReplicatedCassandra {
                                   .getOrElse { from }
                                   .toSegmentNr(segmentSize)
                                   .to[F] {
+                                    // TODO MR for context see test-case `repeat purge again for the same offset`
                                     to
-                                      .next[Option]
+                                      .next[Option] // TODO MR why overshoot to possibly next segment?
```

Comment:
Safety measure?

Reply:
why here we do "safety measure", but not in `def delete` - they both do the same, don't they?

do you think that this is "by design" and not a bug?

Reply:
for me it looks like `off by 1` error

Reply:
I'm afraid but I have no idea _why_ its done like that. On other hand I cannot accept dropping this code without understanding _why_ it was written and _what_ are consequences of a change.